Sources/MessagingPush/MessagingPushImplementation.swift

```diff
@@ -266,3 +311,11 @@ extension MessagingPushImplementation: ProfileIdentifyHook {
         deleteDeviceToken()
     }
 }
+
+extension MessagingPushImplementation: DeviceAttributesHook {
+    // Adds custom device attributes to background queue and sends to workspace
+    func customDeviceAttributesAdded(attributes: [String: Any]) {
+        guard let deviceToken = globalDataStore.pushDeviceToken else { return }
+        addDeviceAttributes(deviceToken: deviceToken, customAttributes: attributes)
```

Comment:
Because custom device attributes are only added when there is a push token, what would happen in this scenario:

* Custom device attributes set via `CustomerIO.instance.deviceAttributes = ["foo": "bar"]`
* No device token registered yet. 
* Device token registered. 

The custom device attributes will never get added. The request to set will get ignored.

Reply:
Yes, you are correct. 

The same issue is present for [profile attributes](https://github.com/customerio/customerio-ios/blob/66f5ec2b7138f5eade7a0c9d732646d2bb7ef1c9/Sources/Tracking/CustomerIOImplementation.swift#L81-L84). So, this shouldn't be a blocker for this PR. I [created a ticket](https://github.com/customerio/issues/issues/6982) to fix attributes for both devices + profile.